packages/babel-plugin-transform-es2015-parameters/src/default.js

```diff
@@ -1,137 +0,0 @@
-import getFunctionArity from "babel-helper-get-function-arity";
```

Comment:
Sorry about the overlap here with https://github.com/babel/babel/pull/5709. I think I may have made some assumptions about how you were approaching this that I didn't communicate well.

I had originally envisioned this functionality as entirely separate from the parameter transform, in the first pass, so allow you to focus on learning the transform API, without having to rewrite all the existing code. For example the same way we have [`arrowToFunctionExpression` here](https://github.com/babel/babel/blob/7.0/packages/babel-traverse/src/path/conversion.js#L55) I was thinking we could have a `fnPath.inlineParametersInBody()` or something, that could be used from `object-rest-spread`. I was thinking a second step would be `fnPath.inlineParametersInBody({disallowRest: true})` could be be used to implement `babel-plugin-transform-es2015-parameters`.

That way you could have focused on fixing the issue with `object-rest-spread` as a first step, without having to change all the logic for `es2015-parameters` like this PR is doing. Also why I hadn't mentioned https://github.com/babel/babel/pull/5709 yet, because I was assuming it would land before you got to this part :P

I didn't do a good enough job communicating all this, sorry.